packages/next/client/page-loader.js

```diff
@@ -111,9 +116,14 @@ export default class PageLoader {
     await this.promisedBuildId
 
     route = this.normalizeRoute(route)
-    const scriptRoute = route === '/' ? '/index.js' : `${route}.js`
+    const scriptRoute = PageLoader.getScriptRoute(route)
 
     const script = document.createElement('script')
+
+    if (process.env.__NEXT_MODERN_BUILD && 'noModule' in script) {
```

Comment:
At this point we already know we're in modern-mode, does the `noModule` property within `script` even matter anymore?

This is dynamically loaded, not placed during initial render.

Reply:
`process.env.__NEXT_MODERN_BUILD` is used to enable disable feature during build time
`'noModule' in script` tells us if the current browser supports `type=module`

Reply:
Right, but the above code always returns the modern bundle if the feature is on, regardless of `'noModule' in script`:
```
js
  static getScriptRoute (route) {
    const scriptName = route === '/' ? '/index' : route
    return `${scriptName}${process.env.__NEXT_MODERN_BUILD ? '.module' : ''}.js`
  }
```